native/cocos/core/scene-graph/Node.cpp

```diff
@@ -475,19 +493,33 @@ void Node::updateWorldTransformRecursive(uint32_t &dirtyBits) { // NOLINT(misc-n
         parent->updateWorldTransformRecursive(dirtyBits);
     }
     dirtyBits |= currDirtyBits;
+    bool positionDirty = dirtyBits & static_cast<uint32_t>(TransformBit::POSITION);
+    bool rotationScaleSkewDirty = dirtyBits & static_cast<uint32_t>(TransformBit::RSS);
     if (parent) {
-        if (dirtyBits & static_cast<uint32_t>(TransformBit::POSITION)) {
+        if (positionDirty && !rotationScaleSkewDirty) {
```

Comment:
Why need to check `!rotationScaleSkewDirty`?

Reply:
See the comment above.

![WeChat519327c773603e5fa1643756781f85d5](https://github.com/user-attachments/assets/f75199bc-bcb4-4969-a379-f014b8c2744b)

Reply:
Sounds reasonable.